types/address.go

```diff
@@ -626,7 +627,17 @@ func Bech32ifyPubKey(pkt Bech32PubKeyType, pubkey crypto.PubKey) (string, error)
 
 	}
 
-	return bech32.ConvertAndEncode(bech32Prefix, legacy.Cdc.MustMarshalBinaryBare(pubkey))
+	// This piece of code is to keep backwards-compatibility.
+	// For ed25519 keys, our own ed25519 is registered in Amino under a
+	// different name than TM's ed25519. But since users are already using
+	// TM's ed25519 bech32 encoding, we explicitly say to bech32-encode our own
+	// ed25519 the same way as TM's ed25519.
+	pkToMarshal := pubkey
+	if ed25519Pk, ok := pubkey.(*ed25519.PubKey); ok {
+		pkToMarshal = ed25519Pk.AsTmPubKey()
+	}
```

Comment:
This piece of code is used to keep backwards-compatibility in the `Bech32ifyPubKey` function.

Bech32ifyPubKey is dependent on the name under which the type has been Amino-registered. The old TM ed25519 pubkey is registered under the same name as before, so our new `ed25519.PubKey` needs to be registered under a new one. But then it's Bech32 pubkey enncoding will change.

Here, we're just saying to bech32ify our ed25519 pubkey as if it were a TM ed22519 pubkey.

---

Talking with @aaronc, a follow-up PR would be to remove `Bech32ifyPubKey` (and all code related to bech32 encoding of a pubkey).

Reply:
Added suggested todo with link to issue.